src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerProfileUpdateService.kt

```diff
@@ -44,7 +51,61 @@ class PrisonerProfileUpdateService(
     }
   }
 
-  private fun Optional<Offender>.orElseThrowNotFound(message: String, prisonerNumber: String) =
+  @Transactional
+  fun updateNationalityOfCurrentBooking(prisonerNumber: String, nationality: String?) {
+    val profileType: ProfileType = profileTypeRepository.nationalityProfile().getOrThrow()
+    val profileCode: ProfileCode? =
+      nationality?.uppercase()?.let { profileCodeRepository.profile(profileType, it).getOrThrow() }
+
+    try {
+      val currentBooking = currentBooking(prisonerNumber)
+      val currentNationality = currentBooking.profileDetails
+        ?.firstOrNull { it.code?.id?.type?.equals(profileType) ?: false }
+
+      if (nationality == null && currentNationality != null) {
+        currentBooking.profileDetails.remove(currentNationality)
+      } else {
+        if (currentNationality != null) {
+          currentNationality.setProfileCode(profileCode)
+        } else {
+          currentBooking.profileDetails.add(
+            OffenderProfileDetail.builder()
+              .id(OffenderProfileDetail.PK(currentBooking, profileType, 1))
+              .caseloadType("INST")
+              .code(profileCode)
+              .listSequence(profileType.listSequence)
+              .build(),
+          )
+        }
+      }
```

Comment:
Could this be simplified by doing:

```
if (currentNationality != null) {
    currentBooking.profileDetails.remove(currentNationality)
}

if (nationality != null) {
    currentBooking.profileDetails.add(
            OffenderProfileDetail.builder()
              .id(OffenderProfileDetail.PK(currentBooking, profileType, 1))
              .caseloadType("INST")
              .code(profileCode)
              .listSequence(profileType.listSequence)
              .build(),
          )
}
```

i.e. not worrying about setting the code on the current row, just get rid of the row and add a new one?

Reply:
No... this causes issues with JPA thinking there's a conflict: `a different object with the same identifier value was already associated with the session`, so ignore this